### ISSUE TITLE
Easy way to run same script on multiple connections?

### ISSUE THREAD

**Issue by Tomikoma** [Tomikoma/dbeaver2#250]
<a href="https://github.com/shlomiw"><img src="https://avatars1.githubusercontent.com/u/1025170?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [shlomiw](https://github.com/shlomiw)**
_Sunday Jun 10, 2018 at 14:24 GMT_
_Originally opened as https://github.com/dbeaver/dbeaver/issues/3605_

----

Hi,
Is it possible to execute same script on multiple different DB connections at once?
i.e.
I have a script and 10 different DB connections (same schema).
Now I have to open SQL script editor tab for each connection, copy+paste the script and execute it.

Any ideas how to expedite this process inside DBeaver?
Thanks

**Comment by Tomikoma** [comment]
<a href="https://github.com/p91paul"><img src="https://avatars2.githubusercontent.com/u/2157006?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [p91paul](https://github.com/p91paul)**
_Sunday Jun 10, 2018 at 18:02 GMT_

----

I usually just change the connection on the top toolbar without creating a new script, this should save you a few clicks.
However the ability to simultaneously execute the same sql on many connections is something I could use too!

**Comment by Tomikoma** [comment]
<a href="https://github.com/shlomiw"><img src="https://avatars1.githubusercontent.com/u/1025170?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [shlomiw](https://github.com/shlomiw)**
_Monday Jun 11, 2018 at 13:01 GMT_

----

Hi @p91paul , tnx for your reply!
The problem is that my scripts are sometimes run for a long time.. so I'd like to execute them in parallel on all DB's.

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Monday Jun 11, 2018 at 17:13 GMT_

----

This makes sense. I think we'll need new UI for that (some special command like "Execute in multiple connections").

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Sunday Jul 15, 2018 at 17:58 GMT_

----

After all I think making just special command isn't good  enough.
Its better to make new "composite" connection. It will contain from multiple other existing connections. Each query executed in this connection will be executed in every "nested" connection.
Basically this is not very hard to do while we do not want to perform cross-database joins and stuff like that.

One of questions is how to represent query results. Say we have composite connection which consists from 3 other connections. E.g. you execute "SELECT * FROM TABLE".
Should it produce 3 tabs for query results? How to visualize this (how to determine which tab belongs to which connection). 

Another question - should it execute all queries in separate threads or execute them one by one?

**Comment by Tomikoma** [comment]
<a href="https://github.com/shlomiw"><img src="https://avatars1.githubusercontent.com/u/1025170?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [shlomiw](https://github.com/shlomiw)**
_Sunday Jul 15, 2018 at 18:34 GMT_

----

Hi, as I see it, simply run the same script in separated tabs in parallel. This what I'm doing now manually.
What do you think?

**Comment by Tomikoma** [comment]
<a href="https://github.com/percyvega"><img src="https://avatars3.githubusercontent.com/u/7452763?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [percyvega](https://github.com/percyvega)**
_Tuesday Jul 17, 2018 at 20:45 GMT_

----

If I could just duplicate the script view, so that I point one view of the script to STG and another view of the same script to PRD, that would solve my issues. This would be similar to what IntelliJ has: "Split Vertically" and "Move Right".

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Wednesday Jul 18, 2018 at 08:08 GMT_

----

@percyvega Regarding UI - you can do the same in DBeaver (main menu Window->Editor->Split editor horizontal/vertical). 
Of course it is not related to multi-connections but you can open two editors with PROD/STG connections and see them on screen at once.

**Comment by Tomikoma** [comment]
<a href="https://github.com/rmarkus"><img src="https://avatars2.githubusercontent.com/u/1353254?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [rmarkus](https://github.com/rmarkus)**
_Monday Feb 18, 2019 at 12:07 GMT_

----

Yes please provide an option to have multi-DB connections. 
The usecase for me is: 
- database with Read-only access (Oracle)
- csv-Data source (or SQLite source) read-write
- requirement to process both, ie for each entry in cvs grab some data from Oracle and write back to a different csv

**Comment by Tomikoma** [comment]
<a href="https://github.com/p91paul"><img src="https://avatars2.githubusercontent.com/u/2157006?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [p91paul](https://github.com/p91paul)**
_Monday Feb 18, 2019 at 18:35 GMT_

----

@rmarkus this ticket is about running the same sql on different connections, not combining data from different sources. What you request would make DBeaver an ETL tool, which is an entirely different software category. You might want to open a new issue with your request, maybe it will be considered.

**Comment by Tomikoma** [comment]
<a href="https://github.com/LucasVandecasteele"><img src="https://avatars0.githubusercontent.com/u/20922910?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [LucasVandecasteele](https://github.com/LucasVandecasteele)**
_Monday Mar 11, 2019 at 16:55 GMT_

----

Hi, I have the exact same need as shlomiw , and i was hoping this feature to be released on 6.0 . Please keep this issue opened

**Comment by Tomikoma** [comment]
<a href="https://github.com/ChaminW"><img src="https://avatars0.githubusercontent.com/u/13014405?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ChaminW](https://github.com/ChaminW)**
_Friday Nov 01, 2019 at 06:12 GMT_

----

Hi, I have the same requirement. Have to execute the same query for multiple slave DB clusters which take significant time.

**Comment by Tomikoma** [comment]
<a href="https://github.com/ChaminW"><img src="https://avatars0.githubusercontent.com/u/13014405?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ChaminW](https://github.com/ChaminW)**
_Friday Nov 01, 2019 at 06:17 GMT_

----

What I suggest is having dynamic data sources for scripts rather having a single active data source. Then we can right query like 
`SELECT * FROM <DATA_SOURCE>.<SCHEMA>.<TABLE> where <SOME_CONDITION>;`

Not sure whether it is feasible. Just suggesting.

**Comment by Tomikoma** [comment]
<a href="https://github.com/dicaetano"><img src="https://avatars3.githubusercontent.com/u/20803573?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dicaetano](https://github.com/dicaetano)**
_Friday Jan 10, 2020 at 18:31 GMT_

----

Please, I have more then 30 connections, and sometimes I have to run a query in all of then. I already love DBeaver, but this would take it to another level.

**Comment by Tomikoma** [comment]
<a href="https://github.com/gabrielbet"><img src="https://avatars1.githubusercontent.com/u/55458497?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [gabrielbet](https://github.com/gabrielbet)**
_Monday Feb 10, 2020 at 13:43 GMT_

----

@dicaetano > "Please, I have more then 30 connections, and sometimes I have to run a query in all of then. I already love DBeaver, but this would take it to another level."

I share the same felling 🙏

**Comment by Tomikoma** [comment]
<a href="https://github.com/rat"><img src="https://avatars2.githubusercontent.com/u/4894037?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [rat](https://github.com/rat)**
_Friday Feb 21, 2020 at 17:21 GMT_

----

Hi, I have the same requirement. I usually need to do an inset or update on a postgres server that has 100 or 200 databases, everyone needs to receive the update or insert

I usually need to open database by database and run, it takes a whole day. the problem is that it needs to be done outside business hours, so that customers are not left without a system.

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Monday Feb 24, 2020 at 16:05 GMT_

----

New SQL script task will support any number of scripts and any number of connections.
Each script will be executed in all selected connections.

Task view->Create New Task->SQL Script.

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Tuesday Feb 25, 2020 at 12:51 GMT_

----

![image](https://user-images.githubusercontent.com/6398845/75249167-bccf0580-57e6-11ea-97e6-9e0d6acbf5db.png)

**Comment by Tomikoma** [comment]
<a href="https://github.com/uslss"><img src="https://avatars0.githubusercontent.com/u/31996417?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [uslss](https://github.com/uslss)**
_Tuesday Feb 25, 2020 at 17:00 GMT_

----

Reopened.
1. Connection could be added more than one time.
2. Scripts folder could be added instead of script, but scripts from folder aren't added
![image](https://user-images.githubusercontent.com/31996417/75268710-ea2cab00-5808-11ea-9742-9a6b09b0ee49.png)
3. Checking/unchecking auto-commit option doesn't have any effect.
 Case: 
    I've got two update script. The second script is incorrect and causes error. Table from first script is always updated (no matter if auto-commit disabled or enabled)
4. "Dump query results to log file" option also have got no effect. Log looks the same
![image](https://user-images.githubusercontent.com/31996417/75269150-822a9480-5809-11ea-8fb5-fa607adf87f9.png)

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Wednesday Feb 26, 2020 at 13:38 GMT_

----

Fixed
